spec/lib/miq_automation_engine/models/miq_ae_export_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+include AutomationSpecHelper
+
+describe MiqAeExport do
+  before do
+    @root_tenant = Tenant.seed
+    @tenant1 = FactoryGirl.create(:tenant)
+    @tenant2 = FactoryGirl.create(:tenant)
+    create_ae_model(:name => "dom1", :tenant => @tenant1)
+    create_ae_model(:name => "dom2", :tenant => @tenant2)
+    @options = {'export_dir' => Dir.tmpdir, 'overwrite' => true}
+  end
+
+  it "export inaccessible domain" do
+    hash = @options.merge('tenant' => @tenant1)
+    expect { MiqAeExport.new('dom2', hash) }.to raise_exception(MiqAeException::DomainNotAccessible)
+  end
+
+  it "export all domains for a tenant" do
+    hash = @options.merge('tenant' => @tenant1)
+    MiqAeExport.stub(:write_domain).and_return("")
+    obj = MiqAeExport.new('*', hash)
+    obj.should_receive(:write_domain).once
+    obj.export
+  end
+
+  it "export a domain as root tenant" do
+    hash = @options.merge('tenant' => @root_tenant)
+    MiqAeExport.stub(:write_domain).and_return("")
+    obj = MiqAeExport.new('dom2', hash)
+    obj.should_receive(:write_domain).once
+    obj.export
+  end
+
+  it "export multiple domain as root tenant" do
+    hash = @options.merge('tenant' => @root_tenant)
+    MiqAeExport.stub(:write_domain).and_return("")
+    obj = MiqAeExport.new('*', hash)
+    obj.should_receive(:write_domain).twice
+    obj.export
+  end
```

Comment:
Think we can combine these tests a little bit better? Maybe something like this:
```ruby
describe MiqAeExport do
  describe "#instantiate" do
    let(:options) { {'export_dir' => Dir.tmpdir, 'overwrite' => true, 'tenant' => tenant} }

    before do
      MiqAeExport.stub(:write_domain).and_return("")
    end

    context "when the tenant is a root tenant" do
      let(:tenant) { Tenant.seed }

      it "exports a single domain" do
        ...
      end

      it "exports multiple domains" do
        ...
      end
    end

    context "when the tenant is not a root tenant" do
      let(:tenant) { FactoryGirl.create(:tenant) }

      it "exports a single domain" do
        ...
      end

      it "blows up exporting an unavailable domain" do
        ....
      end
    end
  end
end
```